### DISCUSSION TITLE
how to install deepin desktop on debian11?

### DISCUSSION THREAD

**Discussion by gnuhub** [linuxdeepin/developer-center#3314] [General | 常规讨论]
# why

1. 起点是选择deepin v23 nightly作为桌面版(和服务器debian11保持一致)
2. 在使用过程中发现 #3313 结果v23 nightly以后不维护，而且以后的版本要脱离debian
3. 既要一个和服务器保持一致的debian11,又要有一个有好的桌面dde
4. 自己动手？还是继续使用xface?

## 已知

1. https://salsa.debian.org/pkg-deepin-team
5. https://ubuntudde.com/
6. https://github.com/linuxdeepin/developer-center/issues/3313
7. https://github.com/linuxdeepin/developer-center/issues/3230

## 未知

1. 如何在debian11安装使用最新版的dde?

## 问题

1. debian移植dde的现状？完成的百分比，参与移植的人员？
2. 有那些任务需要解决？任务列表？
3. 如何开始，如何入手，文档？

**Comment by BLumia** [comment]
需要社区人员移植。有兴趣的话，组个团一起移植？

**Reply by gnuhub** [reply]
# debian的移植
1. 移植的当前进度？了解一下当前完成的百分比，移植的参与人，技术支持？
1. 移植具体有那些任务？需要做什么？看下自己可以做什么？
1. 如何开始 文档 步骤？

**Reply by BLumia** [reply]
这不是还没组成团嘛。我目前只有在尝试处理一些我自行移植过程中遇到的问题。有其它感兴趣的人才能组的起团，然后商讨你提出来的这些事项如何安排。不知道你有没有兴趣 😂 

要做的事情肯定就是让 DDE 在 debian 11 上能跑得起来咯，由于是移植到 debian，所以文档基本是参照 debian 的打包文档。

**Reply by gnuhub** [reply]
# 疑问不解

那个deepin发行的版本不是基于debian开发的吗？怎么会有移植问题？

**Reply by BLumia** [reply]
一些补充：

- 之前有一个 ubuntu 的移植尝试（其实一直到 debian 和 ubuntu 遇到的问题大多是差不多的，以及这个 OBS 仓库也可以加个 debian 源） https://build.opensuse.org/project/show/home:BLumia:dde4u:22.04
- 目前主要的 DDE 项目是有 debian sid 的自动打包的，尽管只能确保包能打过，可参见 https://bbs.deepin.org/post/239588
- 计划上，v23 并不是 debian based，所以 DDE 的 debian 移植可以不考虑 deepin v23 的仓库/计划

**Reply by gnuhub** [reply]
# 组团移植debiandde

1. 可以一起组团移植，可以做个类似ubuntudde的debiandde
2. 主要是如何开始？

**Reply by BLumia** [reply]
> deepin发行的版本不是基于debian开发的吗？怎么会有移植问题？

v20 的基础仓库蛮老的，现在上层应用有的在代码或者工程文件里写了对现有仓库或者环境的一些不正确的依赖，所以会有移植问题。

**Reply by BLumia** [reply]
如果你熟悉 debian 打包的话，可以了解下 OBS 并在那个平台上尝试打包构建 DDE 的基础包（我补充的 dde4u 就是个例子，当然我们也可以一起搞）。遇到问题的话可以开 issue 或者 discussion 讨论，可以发 patch 到项目（我可以协助处理代码评审与合并相关的问题）。如果不熟悉 debian 打包的话，可以先了解下打包相关的知识。熟悉后我们可以考虑组个 SIG 专门处理移植的事情。

另外，ubuntudde 那边好像也打算开始做 DDE 到新 ubuntu LTS 的移植了，不过我不是很了解那边的讨论渠道和进展。

**Comment by gnuhub** [comment]
# 起点问题

## 已知
1. 之前有一个 ubuntu 的移植尝试（其实一直到 debian 和 ubuntu 遇到的问题大多是差不多的，以及这个 OBS 仓库也可以加个 debian 源） https://build.opensuse.org/project/show/home:BLumia:dde4u:22.04
2. 目前主要的 DDE 项目是有 debian sid 的自动打包的，尽管只能确保包能打过，可参见 https://bbs.deepin.org/post/239588

## 问题

这两个对应的源码仓库脚本在哪里？

**Reply by BLumia** [reply]
OBS 的就在你贴的那个链接里。debian sid 自动打包是走的 GitHub Action。参见[这里](https://github.com/linuxdeepin/.github/blob/master/.github/workflows/build-distribution.yml)。

**Comment by gnuhub** [comment]
# debian11安装问题

## debian11安装dde

## 1 添加镜像源
```
 echo "deb [trusted=yes] https://deepin-community.github.io/deepin-dde-repo apricot main" > /etc/apt/sources.list.d/deepin-dde-repo.list
```

## 问题

1. 添加镜像源后，如何安装切换为dde,目前使用的debian 11 xface
2.` apt show dde` 没有一个期望的metapackage全部安装 `apt install deepin-*`? https://github.com/deepin-community/debian-sid-dde-repo/blob/main/dists/sid/main/binary-amd64/Packages
3. 然后下一步有哪些问题？起点在哪里？

**Reply by BLumia** [reply]
first thing first: 上面强调过了，那个仓库是确保**能构建**的，完全**不确保**使用过程没问题。那是逐 commit 仓库，除非你非常清楚你在做什么，否则不建议使用那个仓库作为日常使用。

当你有对应的仓库/包后，然后当然就是尝试安装 DDE 相关的包（没记错的话 deepin 里的 deepin-desktop-base meta 包是这个目的，可以参照它安装），并检查有没有问题了。对于有问题的情况，要做的事情就是处理对应的问题了。

**Comment by gnuhub** [comment]
# TODO

1. debian sid 的自动打包 应该增加一个matapackage https://salsa.debian.org/pkg-deepin-team/dde  ==>  `apt install dde -y` 安装桌面环境
2. 如何切换桌面环境，起点在哪里？

**Reply by BLumia** [reply]
在你的显示管理器里切桌面环境，取决于你用的哪个 displaymanager。

有虚包能确保顺利安装基本并能成功切换到 DDE 就基本已经是终点了，起点是先确保包能打成功，然后是能进入桌面并且工作正常...

**Comment by gnuhub** [comment]
#  求答疑解惑

* deepin v23 nightly,就是基于debian11构建的，为什么v23 nightly桌面正常，而如今在debian11运行dde却需要移植

**Reply by BLumia** [reply]
上面回答过 https://github.com/orgs/linuxdeepin/discussions/3314#discussioncomment-3194338 一次了，尽管是针对 v20 的回答。对于 v23，基础环境并非完全基于 debian 11，并且基础仓库也不会拥有和 debian 11 仓库完全对等数量的包，不应该视为是同一环境。至于为什么需要移植，如果你能直接构建跑起来 DDE 的话，那就不需要移植了。移植的目的就是解决直接构建后尝试运行 DDE 时，与 deepin 相比存在的问题。

**Reply by gnuhub** [reply]
@BLumia 我已经添加了 
```
 echo "deb [trusted=yes] https://deepin-community.github.io/deepin-dde-repo apricot main" > /etc/apt/sources.list.d/deepin-dde-repo.list
```
## 然后我应该如何测试运行DDE，以便于知道存在的问题，给我一个起点

1. 构建我可以看actions 还是说现在构建仓库还构建不出来？有问题？
2. 我在寻找起点，如何启动DDE？

**Reply by BLumia** [reply]
你加的那个不是适用于 debian 的仓库，那个是基于 deepin 的仓库。适用于 debian **sid** 的是 ` https://github.com/deepin-community/debian-sid-dde-repo/ ` 这个。其它版本需要你自己/用打包平台打包...

**Reply by gnuhub** [reply]
这个我可以改成 https://github.com/deepin-community/debian-sid-dde-repo/

# 我的疑问

1. 我如何基于 debian-sid-dde-repo 尝试启动测试DDE?

**Reply by BLumia** [reply]
你的环境是 debian **_sid_**？（ps. 有 telegram 账号吗？可以实时讨论一下，看上去你并不是很熟悉打包相关的过程）

**Reply by gnuhub** [reply]
https://t.me/gnuhub

**Reply by gnuhub** [reply]
我可以下载sid安装一个，只要告诉我如何启动测试

**Comment by BLumia** [comment]
明确回答几个问题：

1. 目前 GitHub Action 跑的 CI 所对应的构建仓库都**不保证**正常运行，只能保证能构建。
2. 现状（2022/07/21）是，无论是目前 GitHub Action 对应的 CI 仓库还是我之前的 OBS 上的仓库，都无法进入 DDE 桌面环境。
3. 移植就是为了解决这些问题。

**Comment by wineee** [comment]
@gnuhub 我在筹建 dde 移植兴趣小组，欢迎来 https://t.me/ddeport 或 https://matrix.to/#/#dde-port:matrix.org 讨论
